src/hooks/vaultsHooks.ts

```diff
@@ -0,0 +1,426 @@
+    queryFn: async () => {
+      const [acc, transfers] = await Promise.all([
+        placeholderFetchVaultAccount(),
+        placeholderFetchVaultAccountTransfers(),
+      ]);
+
+      const parsedAccount = VaultAccountCalculator.getAccountVaultResponse(acc);
+      const parsedTransfers = VaultAccountCalculator.getTransfersBetweenResponse(transfers);
+
+      if (parsedAccount == null || parsedTransfers == null) {
+        return wrapNullable(undefined);
+      }
+      return wrapNullable(
+        VaultAccountCalculator.calculateUserVaultInfo(parsedAccount, parsedTransfers)
+      );
+    },
+    ...vaultQueryOptions,
+  });
+  return accountVault?.data;
+};
+
+export const useLoadedVaultAccountTransfers = () => {
+  const account = useLoadedVaultAccount();
+  return useMemo(() => account?.vaultTransfers?.toArray(), [account?.vaultTransfers]);
+};
+
+const useVaultFormAmountDebounced = () => {
+  const amount = useAppSelector((state) => state.vaults.vaultForm.amount);
+  return useDebounceHf(amount, 500);
+};
+
+export const useVaultFormSlippage = () => {
+  const amount = useVaultFormAmountDebounced();
+  const operation = useAppSelector((state) => state.vaults.vaultForm.operation);
+  const vaultBalance = useLoadedVaultAccount();
+
+  const { data: slippageResp } = useQuery({
+    queryKey: [
+      'vaultSlippage',
+      amount,
+      operation,
+      vaultBalance?.balanceUsdc,
+      vaultBalance?.balanceShares,
+    ],
+    queryFn: async () => {
+      if (operation === 'DEPOSIT' || amount.trim() === '') {
+        return wrapNullable(undefined);
+      }
+      const sharesToWithdraw = MustBigNumber(amount).div(
+        // usdc / share to determine share value
+        MustBigNumber(vaultBalance?.balanceUsdc).div(MustBigNumber(vaultBalance?.balanceShares))
+      );
+      const slippage = await placeholderFetchVaultFormSlippage(
+        sharesToWithdraw.decimalPlaces(0, BigNumber.ROUND_FLOOR).toNumber(),
+        MustBigNumber(amount).times(0.96145).toNumber()
+      );
+
+      const parsedSlippage =
+        VaultDepositWithdrawFormValidator.getVaultDepositWithdrawSlippageResponse(slippage);
+
+      return wrapNullable(parsedSlippage);
+    },
+    ...vaultQueryOptions,
+  });
+  return slippageResp?.data;
+};
+
+const selectVaultFormStateExceptAmount = createAppSelector(
+  [
+    (state) => state.vaults.vaultForm.operation,
+    (state) => state.vaults.vaultForm.slippageAck,
+    (state) => state.vaults.vaultForm.confirmationStep,
+  ],
+  (operation, slippageAck, confirmationStep) => ({
+    operation,
+    slippageAck,
+    confirmationStep,
+  })
+);
+
+const selectSubaccountStateForVaults = createAppSelector(
+  [
+    (state) => state.account.subaccount?.marginUsage?.current,
+    (state) => state.account.subaccount?.freeCollateral?.current,
+    calculateCanViewAccount,
+  ],
+  (marginUsage, freeCollateral, canViewAccount) => ({
+    marginUsage,
+    freeCollateral,
+    canViewAccount,
+  })
```

Comment:
I know it is only being used in this file but could we co-locate this with the other selectors in `vaultSelectors` and `accountSelectors`?